src/lavinmq/exchange/default.cr

```diff
@@ -6,22 +6,30 @@ module LavinMQ
       "direct"
     end
 
-    def bind(destination, routing_key, headers = nil)
-      raise "Access refused"
+    def bindings_details : Iterator(BindingDetails)
+      ([] of BindingDetails).each
     end
 
-    def unbind(destination, routing_key, headers = nil)
-      raise "Access refused"
+    def bindings : Iterator(Destination)
+      ([] of Destination).each
```

Comment:
Nice, thanks

Reply:
Maybe add an `struct EmptyIterator(T)` to get rid of the allocations completely.

I think it could be an improvement to stdlib; add `Iterator(T).empty` that returns an `struct EmptyIterator(T)`.

Reply:
I created an issue for that https://github.com/crystal-lang/crystal/issues/15038